webapp/channels/src/components/edit_post/edit_post.tsx

```diff
@@ -234,12 +236,18 @@ const EditPost = ({editingPost, actions, canEditPost, config, channelId, draft,
             return;
         }
 
-        const updatedPost = {
+        let updatedPost = {
+            ...editingPost.post,
```

Comment:
Yes I remember reading about this somewhere in issues brought up by other encryption plugins. For our use case it would actually be needed for the post.props to be patched as well. Would it be possible that we can modify this PR to include a flag that would default to just updating the message but allow for patching post.props as well if needed?

Reply:
@streamer45 @qryptdev The `runMessageWillBeUpdatedHooks` hook can still edit the `props` right? It's receiving `editingPost.post`, which is the same dataset we'd be providing here.

> Would it be possible that we can modify this PR to include a flag

Are you implying we would had a config setting?

Reply:
@qryptdev Do you mind explaining the different risks and avenues you're aware of here?

Reply:
The main problem we had with post edits containing props in the submission, was that the frontend was overwriting some props data that was generated for client-side use. In particular, the integration post actions are encrypted using a cookie when preparing the post for the client. Essentially the information at `post.props.attachments[0].actions[0].integration` is encrypted into `post.props.attachments[0].actions[0].cookie` and `integration` is removed.

https://github.com/mattermost/mattermost/blob/40ff77c9c6c2148e33a7fc5f49441b00bffd5c4c/server/channels/app/post.go#L566

https://github.com/mattermost/mattermost/blob/40ff77c9c6c2148e33a7fc5f49441b00bffd5c4c/server/public/model/integration_action.go#L339-L343

There is then a mismatch of the exact content in `post.props` on the client, as there is in the database. It seems there is no handling for this case on the server when the update post endpoint is called. The props get updated unconditionally [here](https://github.com/mattermost/mattermost-server/blob/98da282f7cce27375e6e7bc42a3682bdca888fb3/server/channels/app/post.go#L683), and I'm not seeing any checks for ensuring the `action.integration` is still stored correctly.

In order to avoid this situation, we had chosen to make it so the frontend just never sends the props, and so the server just happens to retain the database version's props value upon saving the edited post. I'll do some investigation to see if there needs to be a server-side fix for this, since that is where it seems the issue we're trying to avoid actually lives.

---

In the effort here to reintroduce the `registerMessageWillBeUpdatedHook` hook, I think the solution in my previous comment works well here. It removes the risk for unconditionally overwriting props, and allows a plugin to edit the post's props. There is still risk for a plugin to accidentally edit a post's props that contains a post action cookie, so if you are writing/using a plugin that will edit the props from the client, please keep the `action.integration` in mind if you have a plugin running on your server that relies on post actions.

Reply:
We updated the code per your suggestion. No issues with suggested solution.